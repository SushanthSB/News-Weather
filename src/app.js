import React from 'react'
import Header from './components/header'
import SubHeader from './components/sub-header'
import { connect } from 'react-redux';
import PropTypes from 'prop-types'
import Menu from './components/menu'
import {Weather} from './components/weather'
import { dataFetch, weatherFetch } from './actions/'
import { bindActionCreators } from 'redux'
import SubMenu from './components/sub-menu'
import News from './components/news';

export class App extends React.Component {
  componentDidMount() {
    this.props.weatherFetch() // Call weatherFetch() action to fetch weather on mount
    const devWidth=(this.navElement.clientWidth) //Get navbar width
    const margin = (Math.floor(devWidth/2)-60);//Animate brand name to center on scroll
    if(devWidth > 736) { 
      this.menuBarElement.className=""
      this.menuElement.className="nav nav-tabs"
      window.addEventListener('scroll',() => {
          this.handleScroll(margin)
        })
    }
  }

  handleScroll(margin) {
    if(window.pageYOffset >= 265) {
      this.headerBrandElement.style.marginLeft=margin+"px"
      this.headerBrandElement.style.transition="all 1s"; 
    }
    else {
      this.headerBrandElement.style.marginLeft="150px"
      this.headerBrandElement.style.transition="all 1s"; 
    }
  }
  render() {
    return (
      <div className="App">
        <Header headerBrandRef={(headerBrand) => this.headerBrandElement=headerBrand} navRef={(nav) => this.navElement = nav} />
        <SubHeader news_source={this.props.news_source} />
        <Menu menuRef={(menu) => this.menuElement=menu} menuBar={(menuBar) => this.menuBarElement = menuBar} category={this.props.category} subCategory={this.props.subCategory} handleTerm={this.props.dataFetch} menu={this.props.menu} />
        <SubMenu subCategory={this.props.subCategory} menu={this.props.menu} handleTerm={this.props.dataFetch} />
        <News fetch={this.props.fetch} />
        <Weather weather={this.props.weather} />
      </div>
    );
  }
}

App.propTypes = {
  news_source: PropTypes.array.isRequired,
  category: PropTypes.array.isRequired,
  subCategory: PropTypes.array.isRequired,
  menu: PropTypes.object.isRequired,
  fetch: PropTypes.object.isRequired,
  dataFetch: PropTypes.func,
  weatherFetch: PropTypes.func.isRequired,
  weather: PropTypes.object.isRequired
}

const mapStateToProps = (state) => {
	return {
		news_source: state.news_source,
    category: state.category,
    subCategory: state.subCategory,
    menu: state.menuReducer,
    fetch: state.fetchReducer,
    weather: state.weatherReducer
	}
}

const mapDispatchToProps = (dispatch) => {
  return {
    dataFetch: bindActionCreators(dataFetch, dispatch),
    weatherFetch: bindActionCreators(weatherFetch, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
