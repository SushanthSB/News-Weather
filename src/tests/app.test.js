import React from 'react';
import { App } from '../app'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme';
import { spy } from 'sinon';

describe("App()",() => {
	let props = {
		news_source: ['bbc-news','the-hindhu'],
		category: ['general','business'],
		subCategory:[{'General':['subterm']}], 
       	menu:{'term':'subterm'},
       	dataFetch: spy(),
       	fetch:{data:[{autor:"Jeff"}]},
       	weatherFetch:spy(),
       	weather:{
       		"data":{"log_name":"Bengaluru"},
       		"weather":{"description":"sunny","icon":"04n"},
       		"temp":{"temp":22.45,"temp_max":23,"temp_min":22}
       	}
	}

	it("Should not get modified",() => {
		const tree = renderer.create(<App {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it("Should render without breaking",() => {
		const wrapper = shallow(<App {...props} />);
		expect(wrapper).toHaveLength(1);
	});
	it("ComponentDidMount should get called",() => {
		spy(App.prototype, "componentDidMount")
		mount(<App {...props} />)
		expect(App.prototype.componentDidMount.calledOnce).toEqual(true)
	})
})
