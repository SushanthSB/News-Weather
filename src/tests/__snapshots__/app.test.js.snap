// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App() Should not get modified 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-default navbar-fixed-top header-bar"
  >
    <h1
      className="header-brand"
    >
      <span>
        Daily News
      </span>
      <span>
        <button
          className="navbar-toggle"
          onClick={[Function]}
          type="button"
        >
          <div
            className="icon-bar"
          />
          <div
            className="icon-bar"
          />
          <div
            className="icon-bar"
          />
        </button>
      </span>
    </h1>
  </nav>
  <div
    className="jumbotron sub-header col-xs-12 col-sm-12 col-md-12 col-lg-12"
  >
    <p>
       Get live headlines from 
    </p>
    <h1>
      <div
        className=""
      />
    </h1>
  </div>
  <div
    className="side-nav"
    id="menu-bar"
  >
    <span
      className="side-nav-close"
      onClick={[Function]}
    >
      ×
    </span>
    <ul
      className="menu"
    >
      <li
        className="not-active"
        onClick={[Function]}
      >
        <a />
      </li>
      <li
        className="not-active"
        onClick={[Function]}
      >
        <a />
      </li>
    </ul>
  </div>
  <ul
    className="nav nav-pills submenu"
  />
  <div
    className="panel-group news-group col-xs-12 col-sm-12 col-md-8 col-md-offset-1 col-lg-8 col-lg-offset-1"
  >
    <h3
      className="top-stories"
    >
      Top Stories
    </h3>
    <div
      className="panel panel-default news-default col-xs-12 col-sm-12 col-md-12 col-lg-12"
    >
      <a
        className="col-xs-12 col-sm-12 col-md-12 col-lg-12"
        href={undefined}
      >
        <div
          className="news-img col-xs-2 col-sm-2 col-md-2 col-lg-2"
        >
          <img
            alt={undefined}
            src={undefined}
          />
        </div>
        <div
          className="col-xs-9 col-xs-offset-1 col-sm-9 col-md-9 col-lg-9"
        >
          <div
            className="panel-heading news-title"
          >
             
             
          </div>
          <div
            className="panel-body news-body"
          >
             
             
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="sim"
  >
    <h3
      className="weather-brand"
    >
      Weather
    </h3>
    <div
      className="jumbotron weather-div1 col-md-3 col-lg-3"
    >
      <div
        className="city-name col-xs-12 col-sm-12 col-md-12 col-lg-12"
      >
        <h3 />
      </div>
      <div
        className="weather-div2 col-xs-12 col-sm-12 col-md-12 col-lg-12"
      >
        <table
          className="col-xs-12 col-sm-12 col-md-12 col-lg-12 weather-table"
        >
          <thead>
            <tr>
              <th
                colSpan="2"
              >
                THURSDAY
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="temp"
            >
              <td
                className="temp-img"
                colSpan="2"
              >
                <img
                  alt="sunny"
                  src="http://openweathermap.org/img/w/04n.png"
                />
                <span>
                  22
                </span>
              </td>
            </tr>
            <tr
              className="temp-desc"
            >
              <td
                colSpan="2"
              >
                sunny
              </td>
            </tr>
            <tr
              className="temp-rem"
            >
              <td>
                Temp:
              </td>
              <td>
                22.45
                ℃ | 
                72.41
                ℉
              </td>
            </tr>
            <tr
              className="temp-rem"
            >
              <td>
                Min|Max:
              </td>
              <td>
                22
                ℃ | 
                23
                ℃
              </td>
            </tr>
            <tr
              className="temp-rem"
            >
              <td>
                Min|Max:
              </td>
              <td>
                71.60
                ℉ | 
                73.40
                ℉
              </td>
            </tr>
            <tr
              className="temp-rem"
            >
              <td>
                Humidity
              </td>
              <td>
                %
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
